### PR TITLE
Upgraded latest version of flutter

### PR DESCRIPTION
**Pull request by mevluttosun**
Upgraded android sdk and gradle
Upgraded ios versions

### CODE REVIEW

File: ios/Resources/PrivacyInfo.xcprivacy
Comment (haduyphuong) [inline ios/Resources/PrivacyInfo.xcprivacy:1]: Why you use there permissions?

File: example/pubspec.yaml [hunk 1]
```diff
@@ -1,13 +1,11 @@
 name: flutter_lifecycle_detector_example
-description: A plugin flutter detect app entering background or foreground mode.
-
+description: "Demonstrates how to use the flutter_lifecycle_detector plugin."
 # The following line prevents the package from being accidentally published to
 # pub.dev using `flutter pub publish`. This is preferred for private packages.
-version: 1.0.0+1
 publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 
 environment:
```
Comment (haduyphuong) [inline example/pubspec.yaml:7]: I want the low version to be usable.

File: example/pubspec.yaml [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
-  sdk: '>=2.16.0 <3.0.0'
+  sdk: ^3.5.4
```
Comment (haduyphuong) [inline example/pubspec.yaml]: I want the low version to be usable.

File: ios/Resources/PrivacyInfo.xcprivacy
Comment (mevluttosun) [reply ios/Resources/PrivacyInfo.xcprivacy:1]: It is default. I open default empty plugin then replace it with yours. If you want I can remove.

File: example/.gitignore [hunk 3]
```diff
@@ -1,7 +1,6 @@
 # Miscellaneous
 *.class
 *.log
-*.lock
```
Comment (haduyphuong) [inline example/.gitignore]: I think need keep rule .lock in this file

File: LICENSE [hunk 4]
```diff
@@ -1,20 +1 @@
-The MIT License (MIT)
```
Comment (haduyphuong) [inline LICENSE]: This file should not be deleted or left blank.

File: example/android/app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -31,4 +31,15 @@
             android:name="flutterEmbedding"
             android:value="2" />
     </application>
+    <!-- Required to query activities that can process text, see:
```
Comment (haduyphuong) [inline example/android/app/src/main/AndroidManifest.xml]: Why you can use permission PROCESS_TEXT

File: example/pubspec.lock [hunk 6]
```diff
@@ -0,0 +1,283 @@
+# Generated by pub
```
Comment (haduyphuong) [inline example/pubspec.lock]: I think need remove this file

File: ios/flutter_lifecycle_detector.podspec [hunk 7]
```diff
@@ -5,19 +5,25 @@
 Pod::Spec.new do |s|
   s.name             = 'flutter_lifecycle_detector'
   s.version          = '0.0.1'
-  s.summary          = 'A plugin flutter detect app entering background or foreground mode.'
+  s.summary          = 'A new Flutter plugin project.'
```
Comment (haduyphuong) [inline ios/flutter_lifecycle_detector.podspec]: You was remove description of package????

File: pubspec.yaml [hunk 8]
```diff
@@ -1,20 +1,21 @@
 name: flutter_lifecycle_detector
-description: A plugin flutter detect app entering background or foreground mode.
-version: 0.0.6
-homepage: https://github.com/haduyphuong/flutter_lifecycle_detector
+description: "A new Flutter plugin project."
```
Comment (haduyphuong) [inline pubspec.yaml]: You was remove description of package???? You can not remove link github of this project.

File: pubspec.yaml [hunk 9]
```diff
@@ -1,20 +1,21 @@
 name: flutter_lifecycle_detector
-description: A plugin flutter detect app entering background or foreground mode.
-version: 0.0.6
-homepage: https://github.com/haduyphuong/flutter_lifecycle_detector
```
Comment (haduyphuong) [inline pubspec.yaml]: You was remove description of package????

File: pubspec.yaml [hunk 10]
```diff
@@ -1,20 +1,21 @@
...
+description: "A new Flutter plugin project."
+version: 0.0.1
+homepage:
 
 environment:
-  sdk: '>=2.16.0 <3.0.0'
-  flutter: ">=2.5.0"
+  sdk: ^3.5.4
```
Comment (haduyphuong) [inline pubspec.yaml]: I want the low version to be usable.

**Comment by haduyphuong** [comment]
I was update new sdk Flutter

**Pull request closed without merging**